diff -ur suricata-1.4.1.orig/configure.ac suricata-1.4.1/configure.ac
--- suricata-1.4.1.orig/configure.ac	2013-03-08 07:11:48.000000000 -0500
+++ suricata-1.4.1/configure.ac	2013-03-08 10:45:16.012695864 -0500
@@ -226,10 +226,10 @@
 
     AS_IF([test "x$enable_gccprotect" = "xyes"], [
         #buffer overflow protection
-        AC_MSG_CHECKING(for -fstack-protector)
+        AC_MSG_CHECKING(for -fstack-protector-strong)
         TMPCFLAGS="${CFLAGS}"
-        CFLAGS="${CFLAGS} -fstack-protector"
-        AC_TRY_LINK(,,SECCFLAGS="${SECCFLAGS} -fstack-protector"
+        CFLAGS="${CFLAGS} -fstack-protector-strong"
+        AC_TRY_LINK(,,SECCFLAGS="${SECCFLAGS} -fstack-protector-strong"
         AC_MSG_RESULT(yes),
         AC_MSG_RESULT(no))
         CFLAGS="${TMPCFLAGS}"
